#   치매란?
*   특정 질병이 아닌 정상적으로 생활해오던 사람이 다양한 원인으로 인해 뇌기능이 손상되어 기역력, 사고력, 판단력 등이 일상행활에 지장을 줄 정도로 감소하는 증상을 통칭하는 용어
*   시간이 흐를수록 증상이 점진적으로 악화되는 경향이 있음.
*   알츠하이머병이 치매의 가장 흔한 유형이며, 전체 치매의 약 60-80%를 차지함
*   그 외로는 혈관성 치매, 루이소체 치매, 전측두엽 치매 등이 있음
*   하나만의 원인이 아닌 여러 유형의 치매가 동시에 존재할 수 있음
*   과거에는 노화로 인해 발생하는 질환으로 여겨져왔으나 최근은 유전, 습관, 생활 방식 등 여러 원인이 복합적으로 작용한다고 알려짐
*   진단은 주로 문진 또는 인지 기능 검사로 진행하며, 신체검사/혈액검사(갑상선, 비타민, 전해질 등 이상 확인), 신경영상 검사(CT, MRI 등)를 통하여 진단함
*   증상 완화 약물(콜린성 약물, NMDA 수용체 조절 약물 등)이 존재하며, 초기 및 중기 치매를 완화 시킬수 있음
*   치매의 약 45%는 생활습관 개선 등으로 예방하거나 발병을 지연시킬 수 있음
*   초기 증상으로 기역력 저하, 건방증 등
*   중기 증상으로 언어 구사 문제, 시간/공간 감각 혼란, 판단력 저하, 일상생활 수행 곤란 등
*   말기 증상으로 신체 기능 유지가 어려워져 돌봄이 필요해짐
*   예방은 식습관 개선, 혈압, 혈당, 콜레스테롤 조절, 금주, 금연 등 일상생활 개선이 필요하며
*   신체 활동, 인지 자극 활동, 사회적 활동 등 뇌기능 자극을 통하여 예방 또는 발병지연이 가능하다.

### 참고문헌

1.   [Centers for Disease Control and Prevention(CDC, 미국 질병통제예방센터): Alzheimer's Disease and Dementia](https://www.cdc.gov/alzheimers-dementia/about/)
2.   [서울대학교병원 의학정보: 치매](https://terms.naver.com/entry.naver?docId=927317&cid=51007&categoryId=51007)

In [71]:
import pandas as pd
import nanpy as np
import matplotlib as plt
import seaborn as sns

### 같은 이름의 3개의 파일이 같은 파일인지 확인

In [72]:
# tr_lab_df1 = pd.read_csv(r"1.Training\라벨링데이터\1.걸음걸이\training_label.csv")
# tr_lab_df2 = pd.read_csv(r"1.Training\라벨링데이터\2.수면\training_label.csv")
# tr_lab_df3 = pd.read_csv(r"1.Training\라벨링데이터\3.인지기능\training_label.csv")

In [73]:
# all_equal = tr_lab_df1.equals(tr_lab_df2) and tr_lab_df2.equals(tr_lab_df3)
# print(all_equal)  

### 데이터 확인

In [74]:
tr_lab_df = pd.read_csv(r"1.Training\라벨링데이터\1.걸음걸이\training_label.csv")

### SAMPLE_EMAIL
*   각 환자들의 이메일, ID로 사용될 수 있음

### DIAG_NM
*   각 환자들의 상태를 라벨링 한 데이터로 확인됨
*   CN(Cognitively Normal): 정상 인지 상태
*   MCI(Mild Cognitive Impairment): 경도 인지 장애
*   Dem(Dementia): 치매

##### 참고문헌
*   [Dementia and Cognitive Impairment: Epidemiology, Diagnosis, and Treatment](https://pmc.ncbi.nlm.nih.gov/articles/PMC4104432/)

In [75]:
tr_lab_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 141 entries, 0 to 140
Data columns (total 2 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   SAMPLE_EMAIL  141 non-null    object
 1   DIAG_NM       141 non-null    object
dtypes: object(2)
memory usage: 2.3+ KB


In [76]:
tr_lab_df['DIAG_NM'].unique()

array(['CN', 'MCI', 'Dem'], dtype=object)

In [77]:
tr_act_df = pd.read_csv(r"1.Training\원천데이터\1.걸음걸이\train_activity.csv")
tr_sle_df = pd.read_csv(r"1.Training\원천데이터\2.수면\train_sleep.csv")
tr_mmse_df = pd.read_csv(r"1.Training\원천데이터\3.인지기능\train_mmse.csv")

##  train_actovoty
*   각 환자들의 일별 활동 정보를 
*   141명의 actovoty정보가 들어있음
*   activity_day_end, activity_day_start: 시계열 데이터, 활동 시작 및 종료 시간 
*   activity_class_5min, activity_met_1min: ...으로 비어있는 데이터 데이터베이스에서 BLOB타입 데이터를 꺼내면서 소실된 것으로 확인
*   CONVERT(activity_class_5min USING utf8), CONVERT(activity_met_1min USING utf8): 소실된 BLOB타입 데이터를 UTF8으로 컨버팅하여 꺼낸 데이터로 확인
*   activity_class_5min: 하루 5분 당 심박동 로그로 5분단위로 심박수 상태를 1, 2, 3, 4로 저장
*   activity_met_1min: 하루 1분당 MET로그
*   MET란 Metabolic Equivalent of Task[by CDC](https://www.cdc.gov/physical-activity-basics/measuring/index.html?utm_source=chatgpt.com)의 약자로 신체 활동의 강도를 나타내는 단위
*   안정시의 에너지 소모량을 1로 1MET는 3.5ml O₂/kg/min 정도 된다
*   1MET는 가만히 앉아있는 소모데는 에너지 또는 산소량
*   3~5.9MET는 중강도 신체활동
*   6MET 이상은 고강도 신체활동을 뜻한다. 

In [78]:
tr_act_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9705 entries, 0 to 9704
Data columns (total 31 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   EMAIL                                    9705 non-null   object 
 1   activity_average_met                     9705 non-null   float64
 2   activity_cal_active                      9705 non-null   int64  
 3   activity_cal_total                       9705 non-null   int64  
 4   activity_class_5min                      9705 non-null   object 
 5   activity_daily_movement                  9705 non-null   int64  
 6   activity_day_end                         9705 non-null   object 
 7   activity_day_start                       9705 non-null   object 
 8   activity_high                            9705 non-null   int64  
 9   activity_inactive                        9705 non-null   int64  
 10  activity_inactivity_alerts               9705 no

In [79]:
len(tr_act_df['EMAIL'].unique())

141

In [80]:
tr_act_df.head()

EMAIL  activity_average_met  activity_cal_active  \
0  nia+279@rowan.kr               1.28125                  196   
1  nia+279@rowan.kr               1.25000                  145   
2  nia+279@rowan.kr               1.21875                  118   
3  nia+279@rowan.kr               1.28125                  180   
4  nia+279@rowan.kr               1.46875                  374   

   activity_cal_total activity_class_5min  activity_daily_movement  \
0                2251                 ...                     3353   
1                2159                 ...                     2516   
2                2140                 ...                     1716   
3                2240                 ...                     2791   
4                2559                 ...                     5393   

            activity_day_end         activity_day_start  activity_high  \
0  2020-10-20T03:59:59+09:00  2020-10-19T04:00:00+09:00              0   
1  2020-10-21T03:59:59+09:00  2020-10-20T04:00:00+09:00              0   
2  2020-10-22T03:59:59+09:00  2020-10-21T04:00:00+09:00              0   
3  2020-10-23T03:59:59+09:00  2020-10-22T04:00:00+09:00              0   
4  2020-10-24T03:59:59+09:00  2020-10-23T04:00:00+09:00              0   

   activity_inactive  ...  activity_score_meet_daily_targets  \
0                714  ...                                 78   
1                719  ...                                 60   
2                708  ...                                 43   
3                705  ...                                 43   
4                437  ...                                 43   

   activity_score_move_every_hour  activity_score_recovery_time  \
0                             100                           100   
1                             100                           100   
2                              95                           100   
3                             100                           100   
4                             100                           100   

  activity_score_stay_active  activity_score_training_frequency  \
0                         54                                100   
1                         58                                 96   
2                         55                                 71   
3                         58                                 71   
4                         89                                 40   

   activity_score_training_volume  activity_steps  activity_total  \
0                              97            4212             207   
1                              96            3306             140   
2                              83            2550             131   
3                              59            4009             208   
4                              46            8081             517   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            CONVERT(activity_class_5min USING utf8)  \
0  1/1/2/2/2/2/1/1/1/1/1/1/1/2/2/2/2/2/2/3/3/2/2/3/3/3/2/2/2/2/2/3/3/3/2/2/2/2/2/2/3/4/3/2/2/2/2/2/1/1/2/2/2/2/2/2/3/4/2/2/2/2/2/3/4/4/2/2/3/2/2/2/2/2/2/2/2/1/1/2/2/2/3/4/4/3/3/2/3/4/3/3/2/2/2/2/3/2/3/2/2/1/1/1/2/2/2/2/2/2/2/2/2/3/2/2/2/2/2/2/2/2/2/2/2/1/1/2/2/2/2/3/2/2/3/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/3/3/3/2/2/2/2/2/2/3/4/3/3/3/3/3/2/2/2/3/3/3/3/2/2/2/2/2/2/2/2/2/2/1/2/1/1/1/2/2/2/2/2/2/2/2/2/2/1/1/1/2/2/2/2/2/3/2/3/2/2/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/1/1/1/2/1/1/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1

In [81]:
tr_act_df.columns

Index(['EMAIL', 'activity_average_met', 'activity_cal_active',
       'activity_cal_total', 'activity_class_5min', 'activity_daily_movement',
       'activity_day_end', 'activity_day_start', 'activity_high',
       'activity_inactive', 'activity_inactivity_alerts', 'activity_low',
       'activity_medium', 'activity_met_1min', 'activity_met_min_high',
       'activity_met_min_inactive', 'activity_met_min_low',
       'activity_met_min_medium', 'activity_non_wear', 'activity_rest',
       'activity_score', 'activity_score_meet_daily_targets',
       'activity_score_move_every_hour', 'activity_score_recovery_time',
       'activity_score_stay_active', 'activity_score_training_frequency',
       'activity_score_training_volume', 'activity_steps', 'activity_total',
       'CONVERT(activity_class_5min USING utf8)',
       'CONVERT(activity_met_1min USING utf8)'],
      dtype='object')

In [82]:
obj_list = ['activity_class_5min', 'activity_day_end', 'activity_day_start', 'activity_met_1min', 'CONVERT(activity_class_5min USING utf8)', 'CONVERT(activity_met_1min USING utf8)']
tr_act_df[obj_list[:3]]

activity_class_5min           activity_day_end         activity_day_start
0                    ...  2020-10-20T03:59:59+09:00  2020-10-19T04:00:00+09:00
1                    ...  2020-10-21T03:59:59+09:00  2020-10-20T04:00:00+09:00
2                    ...  2020-10-22T03:59:59+09:00  2020-10-21T04:00:00+09:00
3                    ...  2020-10-23T03:59:59+09:00  2020-10-22T04:00:00+09:00
4                    ...  2020-10-24T03:59:59+09:00  2020-10-23T04:00:00+09:00
...                  ...                        ...                        ...
9700                 ...  2020-12-25T03:59:59+09:00  2020-12-24T04:00:00+09:00
9701                 ...  2020-12-27T03:59:59+09:00  2020-12-26T04:00:00+09:00
9702                 ...  2020-12-28T03:59:59+09:00  2020-12-27T04:00:00+09:00
9703                 ...  2020-12-29T03:59:59+09:00  2020-12-28T04:00:00+09:00
9704                 ...  2020-12-30T03:59:59+09:00  2020-12-29T04:00:00+09:00

[9705 rows x 3 columns]

In [83]:
print(tr_act_df['activity_day_end'].unique())
print(len(tr_act_df['activity_day_end'].unique()))

['2020-10-20T03:59:59+09:00' '2020-10-21T03:59:59+09:00'
 '2020-10-22T03:59:59+09:00' '2020-10-23T03:59:59+09:00'
 '2020-10-24T03:59:59+09:00' '2020-10-25T03:59:59+09:00'
 '2020-10-26T03:59:59+09:00' '2020-10-27T03:59:59+09:00'
 '2020-10-28T03:59:59+09:00' '2020-10-29T03:59:59+09:00'
 '2020-10-30T03:59:59+09:00' '2020-10-31T03:59:59+09:00'
 '2020-11-01T03:59:59+09:00' '2020-11-02T03:59:59+09:00'
 '2020-11-03T03:59:59+09:00' '2020-11-04T03:59:59+09:00'
 '2020-11-05T03:59:59+09:00' '2020-11-06T03:59:59+09:00'
 '2020-11-07T03:59:59+09:00' '2020-11-08T03:59:59+09:00'
 '2020-11-09T03:59:59+09:00' '2020-11-10T03:59:59+09:00'
 '2020-11-11T03:59:59+09:00' '2020-11-12T03:59:59+09:00'
 '2020-11-13T03:59:59+09:00' '2020-11-14T03:59:59+09:00'
 '2020-11-16T03:59:59+09:00' '2020-11-17T03:59:59+09:00'
 '2020-11-18T03:59:59+09:00' '2020-11-19T03:59:59+09:00'
 '2020-11-20T03:59:59+09:00' '2020-11-21T03:59:59+09:00'
 '2020-11-22T03:59:59+09:00' '2020-11-23T03:59:59+09:00'
 '2020-11-24T03:59:59+09:00' '2

In [84]:
tr_act_df[obj_list[3:]]

activity_met_1min  \
0                  ...   
1                  ...   
2                  ...   
3                  ...   
4                  ...   
...                ...   
9700               ...   
9701               ...   
9702               ...   
9703               ...   
9704               ...   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               CONVERT(activity_class_5min USING utf8)  \
0     1/1/2/2/2/2/1/1/1/1/1/1/1/2/2/2/2/2/2/3/3/2/2/3/3/3/2/2/2/2/2/3/3/3/2/2/2/2/2/2/3/4/3/2/2/2/2/2/1/1/2/2/2/2/2/2/3/4/2/2/2/2/2/3/4/4/2/2/3/2/2/2/2/2/2/2/2/1/1/2/2/2/3/4/4/3/3/2/3/4/3/3/2/2/2/2/3/2/3/2/2/1/1/1/2/2/2/2/2/2/2/2/2/3/2/2/2/2/2/2/2/2/2/2/2/1/1/2/2/2/2/3/2/2/3/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/3/3/3/2/2/2/2/2/2/3/4/3/3/3/3/3/2/2/2/3/3/3/3/2/2/2/2/2/2/2/2/2/2/1/2/1/1/1/2/2/2/2/2/2/2/2/2/2/1/1/1/2/2/2/2/2/3/2/3/2/2/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/1/1/1/2/1/1/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/1/1/1/1/1/   
1     1/1/1/2/2/2/2/2/2/2/2/2/1/1/1/1/1/2/3/3/3/3/2/2/3/2/2/2/2/3/2/2/2/2/2/2/1/2/1/1/1/2/2/2/3/1/2/2/2/1/2/3/2/2/3/3/2/2/2/2/2/2/1/1/1/1/2/2/1/1/2/2/3/3/4/3/3/3/2/3/2/2/3/3/3/2/2/2/2/3/2/2/2/2/2/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/2/1/1/2/2/2/2/2/2/3/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/3/4/4/4/2/1/2/2/2/2/2/2/2/2/3/3/3/2/3/3/3/2/2/2/2/2/2/2/2/2/2/2/1/1/2/2/2/2/1/2/2/2/2/2/2/2/2/2/2/2/2/2/3/2/2/2/1/1/1/1/2/1/1/2/1/1/1/2/2/1/1/2/2/2/2/3/2/2/2/2/2/2/2/2/1/1/1/1/1/2/2/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/2/2/2/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/2/2/2/2/1/1/1/1/1/   
2     1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/3/3/3/3/3/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/1/2/2/2/2/1/2/2/1/2/1/1/1/1/1/2/2/2/2/2/2/2/2/2/2/2/2/3/3/3/2/3/4/4/3/3/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/1/1/1/1/1/1/2/2/2/2/2/2/2/2/2/2/2/3/3/2/2/2/2/3/2/3/3/2/2/2/2/2/1/2/2/1/1/1/1/1/1/2/2/2/2/2/2/3/2/3/3/3/2/3/2/3/2/2/2/2/3/3/2/2/2/2/2/2/2/2/3/3/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/1/1/1/2/2/2/1/1/2/2/2/2/2/1/2/2/2/2/2/2/2/2/2/2/1/1/1/1/1/1/1/1/1/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/2/2/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/   
3     1/1/1/1/1/1/1/1/1/1/2/2/2/2/3/3/3/2/3/3/3/3/3/2/3/2/3/3/3/2/2/3/2/3/3/3/2/2/2/2/2/2/2/3/2/2/2/2/2/2/2/2/2/3/2/3/3/3/4/3/3/4/3/3/3/3/3/3/2/2/2/2/2/3/4/3/4/4/3/3/3/2/3/2/2/2/2/2/2/1/2/2/2/2/2/3/3/2/3/2/2/2/2/2/2/2/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/3/2/2/2/2/2/2/2/1/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/1/1/2/2/2/2/2/2/2/1/1/1/2/2/3/2/2/2/2/2/2/2/2/3/3/3/2/2/2/2/2/2/2/2/2/1/1/2/2/2/2/2/2/2/2/2/2/2/2/2/1/1/1/1/1/1/1/3/3/2/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/1/1/1/1/1/2/2/1/1/1/1/1/1/1/1/1/1/2/1/1/1/1/1/2/2/2/2/2/1/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/1/1/1/1/1/1/1/   
4     1/1/1/2/2/2/2/2/2/1/1/1/3/2/2/1/2/2/2/3/3/2/2/2/2/2/2/2/2/2/2/2/2/2/2/1/1/2/2/3/3/3/3/3/3/3/3/2/3/3/3/3/3/3/3/2/3/3/3/3/2/3/3/3/2/1/0/0/0/0/0/0/3/3/2/2/2/2/2/2/2/2/2/2/2/2/2/2/3/3/2/2/3/3/3/2/3/2/3/3/3/2/3/2/3/2/2/3/2/3/2/3/2/3/2/2/4/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/2/3/3/3/3/3/3/2/3/3/3/3/3/3/3/3/2/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/2/3/3/3/2/2/2/2/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/3/2/2/3/3/2/2/2/3/2/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/   
...                                                                                                                                                                             

In [85]:
tr_act_df[['activity_class_5min', 'activity_met_1min']]

activity_class_5min activity_met_1min
0                    ...               ...
1                    ...               ...
2                    ...               ...
3                    ...               ...
4                    ...               ...
...                  ...               ...
9700                 ...               ...
9701                 ...               ...
9702                 ...               ...
9703                 ...               ...
9704                 ...               ...

[9705 rows x 2 columns]

In [86]:
print(tr_act_df['activity_class_5min'].unique(), tr_act_df['activity_met_1min'].unique())

['...'] ['...']


In [87]:
pd.set_option("display.max_colwidth", None) 
tr_act_df['activity_class_5min']

0       ...
1       ...
2       ...
3       ...
4       ...
       ... 
9700    ...
9701    ...
9702    ...
9703    ...
9704    ...
Name: activity_class_5min, Length: 9705, dtype: object

In [88]:
tr_act_df['CONVERT(activity_class_5min USING utf8)'][0]

'1/1/2/2/2/2/1/1/1/1/1/1/1/2/2/2/2/2/2/3/3/2/2/3/3/3/2/2/2/2/2/3/3/3/2/2/2/2/2/2/3/4/3/2/2/2/2/2/1/1/2/2/2/2/2/2/3/4/2/2/2/2/2/3/4/4/2/2/3/2/2/2/2/2/2/2/2/1/1/2/2/2/3/4/4/3/3/2/3/4/3/3/2/2/2/2/3/2/3/2/2/1/1/1/2/2/2/2/2/2/2/2/2/3/2/2/2/2/2/2/2/2/2/2/2/1/1/2/2/2/2/3/2/2/3/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/2/3/3/3/2/2/2/2/2/2/3/4/3/3/3/3/3/2/2/2/3/3/3/3/2/2/2/2/2/2/2/2/2/2/1/2/1/1/1/2/2/2/2/2/2/2/2/2/2/1/1/1/2/2/2/2/2/3/2/3/2/2/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/1/1/1/2/1/1/1/2/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/1/1/1/1/1/'

In [89]:
def convert_int(df):
    df = str(df).strip('/')
    return [int(v) for v in df.split('/') if v]

def convert_flo(df):
    df = str(df).strip('/')
    return [float(v) for v in df.split('/') if v]

In [90]:
tr_act_df['activity_class_5min_list'] = tr_act_df['CONVERT(activity_class_5min USING utf8)'].apply(convert_int)
tr_act_df['activity_met_1min_list'] = tr_act_df['CONVERT(activity_met_1min USING utf8)'].apply(convert_flo)


In [91]:
tr_act_df[['activity_class_5min_list','activity_met_1min_list']].head(1)

activity_class_5min_list  \
0  [1, 1, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 3, 3, 2, 2, 3, 3, 3, 2, 2, 2, 2, 2, 3, 3, 3, 2, 2, 2, 2, 2, 2, 3, 4, 3, 2, 2, 2, 2, 2, 1, 1, 2, 2, 2, 2, 2, 2, 3, 4, 2, 2, 2, 2, 2, 3, 4, 4, 2, 2, 3, 2, 2, 2, 2, 2, 2, 2, 2, 1, 1, 2, 2, 2, 3, 4, 4, 3, 3, 2, 3, 4, 3, 3, 2, 2, 2, 2, 3, 2, 3, 2, ...]   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      activity_met_1min_list  
0  [1.2, 0.9, 1.0, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 1.2, 0.9, 1.2, 1.2, 1.2, 1.2, 1.2, 1.2, 1.2, 1.2, 1.1, 1.2, 1.1, 1.2, 1.3, 1.2, 1.2, 1.2, 1.1, 0.9, 0.9, 0.9, 1.0, 1.0, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 1.2, 0.9, 0.9, 0.9, 1.0, 0.9, 0.9, 0.9, 0.9, 1.0, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 1.3, 1.3, 1.4, 1.3, 1.4, 1.2, 1.4, 1.4, 1.2, 1.3, 1.3, 1.2, 1.2, 1.2, 1.1, 1.2, 1.3, 1.3, 1.2, 1.2, 1.3, 1.2, 1.2, 1.1, 1.5, 1.3, 1.4, 1.4, 1.3, 1.3, 1.2, 1.6, ...]

In [92]:
len(tr_act_df['activity_class_5min_list'][0])

288

In [93]:
set(tr_act_df['activity_class_5min_list'][0])

{1, 2, 3, 4}

In [94]:
len(tr_act_df['activity_met_1min_list'][0])

1440

In [95]:
set(tr_act_df['activity_met_1min_list'][0])

{0.9,
 1.0,
 1.1,
 1.2,
 1.3,
 1.4,
 1.5,
 1.6,
 1.7,
 1.8,
 1.9,
 2.0,
 2.1,
 2.2,
 2.3,
 2.4,
 2.5,
 2.6,
 2.7,
 2.8,
 2.9,
 3.0,
 3.1,
 3.2,
 3.4,
 3.5,
 3.6,
 3.7,
 3.8,
 3.9,
 4.0,
 4.1,
 4.2,
 4.4,
 4.7,
 4.8,
 5.1,
 5.3}

In [109]:
tr_act_df['activity_day_end'] = pd.to_datetime(tr_act_df['activity_day_end'], errors='coerce')

tr_act_df['end_year'] = tr_act_df['activity_day_end'].dt.year
tr_act_df['end_month'] = tr_act_df['activity_day_end'].dt.month
tr_act_df['end_day'] = tr_act_df['activity_day_end'].dt.day
tr_act_df['end_hour'] = tr_act_df['activity_day_end'].dt.hour

tr_act_df['activity_day_start'] = pd.to_datetime(tr_act_df['activity_day_start'], errors='coerce')

tr_act_df['sta_year'] = tr_act_df['activity_day_start'].dt.year
tr_act_df['sta_month'] = tr_act_df['activity_day_start'].dt.month
tr_act_df['sta_day'] = tr_act_df['activity_day_start'].dt.day
tr_act_df['sta_hour'] = tr_act_df['activity_day_start'].dt.hour

In [111]:
tr_act_df[['end_year','end_month','end_day','end_hour','sta_year','sta_month','sta_day','sta_hour']].head()

end_year  end_month  end_day  end_hour  sta_year  sta_month  sta_day  \
0      2020         10       20         3      2020         10       19   
1      2020         10       21         3      2020         10       20   
2      2020         10       22         3      2020         10       21   
3      2020         10       23         3      2020         10       22   
4      2020         10       24         3      2020         10       23   

   sta_hour  
0         4  
1         4  
2         4  
3         4  
4         4

In [112]:
tr_act_df.columns

Index(['EMAIL', 'activity_average_met', 'activity_cal_active',
       'activity_cal_total', 'activity_class_5min', 'activity_daily_movement',
       'activity_day_end', 'activity_day_start', 'activity_high',
       'activity_inactive', 'activity_inactivity_alerts', 'activity_low',
       'activity_medium', 'activity_met_1min', 'activity_met_min_high',
       'activity_met_min_inactive', 'activity_met_min_low',
       'activity_met_min_medium', 'activity_non_wear', 'activity_rest',
       'activity_score', 'activity_score_meet_daily_targets',
       'activity_score_move_every_hour', 'activity_score_recovery_time',
       'activity_score_stay_active', 'activity_score_training_frequency',
       'activity_score_training_volume', 'activity_steps', 'activity_total',
       'CONVERT(activity_class_5min USING utf8)',
       'CONVERT(activity_met_1min USING utf8)', 'activity_class_5min_list',
       'activity_met_1min_list', 'end_year', 'end_month', 'end_day',
       'end_hour', 'sta_year', 's

In [119]:

diff_df = tr_act_df[[tr_act_df for tr_act_df in tr_act_df.columns[:-10] if tr_act_df not in obj_list]]
diff_df.columns

Index(['EMAIL', 'activity_average_met', 'activity_cal_active',
       'activity_cal_total', 'activity_daily_movement', 'activity_high',
       'activity_inactive', 'activity_inactivity_alerts', 'activity_low',
       'activity_medium', 'activity_met_min_high', 'activity_met_min_inactive',
       'activity_met_min_low', 'activity_met_min_medium', 'activity_non_wear',
       'activity_rest', 'activity_score', 'activity_score_meet_daily_targets',
       'activity_score_move_every_hour', 'activity_score_recovery_time',
       'activity_score_stay_active', 'activity_score_training_frequency',
       'activity_score_training_volume', 'activity_steps', 'activity_total'],
      dtype='object')

In [120]:
df = tr_act_df.drop(['activity_met_1min','activity_class_5min','CONVERT(activity_class_5min USING utf8)', 'CONVERT(activity_met_1min USING utf8)'], axis=1)

In [ ]:
tr_act_df_v1 = pd.read_csv(r'1.Training\원천데이터\1.걸음걸이\train_activity_v1.csv')

In [123]:
def upseampl(df_list):
    return np.repeat(df_list, 5)[:1440]

def BLOB_to_vecter(df, col_heart, col_MET):
    heart = np.array(df[col_heart], dtype='int')
    met = np.array(df[col_MET], dtype='float')

    if len(met) == len(heart) * 5:
        heart = upseampl(heart)
    elif len(met) != len(heart) * 5:
        heart = upseampl(heart)[:len(met)]

    return heart, met

In [125]:
heart_vec, met_vec = [], []

for _, row in tr_act_df.iterrows():
    heart, met = BLOB_to_vecter(row, 'activity_class_5min_list','activity_met_1min_list')
    heart_vec.append(heart)
    met_vec.append(met)

tr_act_df['heart_vecter'] = heart_vec
tr_act_df['met_vecter'] = met_vec

AttributeError: module 'nanpy' has no attribute 'array'

In [105]:
df.to_csv(r"1.Training\원천데이터\1.걸음걸이\train_activity_v1.csv")

In [96]:
tr_sle_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9705 entries, 0 to 9704
Data columns (total 36 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   EMAIL                                     9705 non-null   object 
 1   sleep_awake                               9705 non-null   int64  
 2   sleep_bedtime_end                         9705 non-null   object 
 3   sleep_bedtime_start                       9705 non-null   object 
 4   sleep_breath_average                      9705 non-null   float64
 5   sleep_deep                                9705 non-null   int64  
 6   sleep_duration                            9705 non-null   int64  
 7   sleep_efficiency                          9705 non-null   int64  
 8   sleep_hr_5min                             9705 non-null   object 
 9   sleep_hr_average                          9705 non-null   float64
 10  sleep_hr_lowest                     

In [97]:
len(tr_sle_df['EMAIL'].unique())

141

In [ ]:
tr_sle_df[['sleep_temperature_delta', 'sleep_temperature_deviation']]

sleep_temperature_delta  sleep_temperature_deviation
0                        0.00                         0.00
1                        0.34                         0.34
2                       -0.07                        -0.07
3                       -0.14                        -0.14
4                       -0.50                        -0.50
...                       ...                          ...
9700                    -0.09                        -0.09
9701                     0.41                         0.41
9702                     0.27                         0.27
9703                    -0.05                        -0.05
9704                     0.26                         0.26

[9705 rows x 2 columns]

In [ ]:
all_equal = tr_sle_df['sleep_temperature_delta'].equals(tr_sle_df['sleep_temperature_deviation']) 
print(all_equal)  

True


In [15]:
tr_mmse_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 141 entries, 0 to 140
Data columns (total 38 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   SAMPLE_EMAIL  141 non-null    object
 1   DIAG_SEQ      141 non-null    int64 
 2   DIAG_NM       141 non-null    object
 3   DOCTOR_NM     141 non-null    object
 4   MMSE_NUM      141 non-null    int64 
 5   MMSE_KIND     141 non-null    int64 
 6   Q01           141 non-null    int64 
 7   Q02           141 non-null    int64 
 8   Q03           141 non-null    int64 
 9   Q04           141 non-null    int64 
 10  Q05           141 non-null    int64 
 11  Q06           141 non-null    int64 
 12  Q07           141 non-null    int64 
 13  Q08           141 non-null    int64 
 14  Q09           141 non-null    int64 
 15  Q10           141 non-null    int64 
 16  Q11_1         141 non-null    int64 
 17  Q11_2         141 non-null    int64 
 18  Q11_3         141 non-null    int64 
 19  Q12_1   